apps/myaccount/src/components/account-recovery/options/sms-recovery/sms-recovery.tsx

```diff
@@ -0,0 +1,429 @@
+                    ),
+                    level: AlertLevels.SUCCESS,
+                    message: t(
+                        "myAccount:components.accountRecovery.SMSRecovery.notifications.updateMobile.success.message"
+                    )
+                });
+
+                dispatch(getProfileInformation());
+                dispatch(setActiveForm(null));
+            })
+            .catch((error: AxiosError) => {
+                if (error.response && error.response.data && error.response.data.detail) {
+                    onAlertFired({
+                        description: t(
+                            "myAccount:components.accountRecovery.SMSRecovery." +
+                            "notifications.updateMobile.error.description",
+                            { description: error.response.data.detail }
+                        ),
+                        level: AlertLevels.ERROR,
+                        message: t(
+                            "myAccount:components.accountRecovery.SMSRecovery." +
+                            "notifications.updateMobile.error.message"
+                        )
+                    });
+
+                    return;
+                }
+
+                onAlertFired({
+                    description: t(
+                        "myAccount:components.accountRecovery.SMSRecovery." +
+                        "notifications.updateMobile.genericError.description"
+                    ),
+                    level: AlertLevels.ERROR,
+                    message: t(
+                        "myAccount:components.accountRecovery.SMSRecovery." +
+                        "notifications.updateMobile.genericError.message"
+                    )
+                });
+            });
+    };
+
+    /**
+     * This function gets the mobile number from the response passed as the argument
+     * and assigns it to mobile and editedMobile.
+     * @param response - response as the parameter.
+     * Temporarily the first mobile type element in the phoneNumbers array is shown.
+     * In the future, we need to decide whether or not to allow multiple recovery mobiles.
+     */
+    const setMobileNumber = (response: BasicProfileInterface) => {
+        let mobileNumber: string = "";
+
+        if (response.phoneNumbers) {
+            if (typeof response.phoneNumbers[0] === "object") {
+                const mobileValue : MultiValue = response.phoneNumbers.find(
+                    (phoneNumber : MultiValue) => phoneNumber.type === "mobile");
+
+                mobileNumber = mobileValue?.value;
+                mobileType = mobileValue?.type;
+            } else {
+                mobileNumber = response.phoneNumbers[0] as string;
+                mobileType = "array";
+            }
+        }
+        setMobile(mobileNumber);
+        setEditedMobile(mobileNumber);
+    };
+
+    useEffect(() => {
+        if (!isEmpty(profileInfo)) {
+            setMobileNumber(profileInfo);
```

Comment:
Is it intentional to set whole profile info as mobile number?

Reply:
This is not setting the whole profile. The function being called is the one from the earlier comment which receives the profileInfo and sets the mobile number using that.